product_dimension/views/product_view.xml

```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <record model="ir.ui.view" id="product_normal_form_view">
+      <field name="name">product_normal_form_view</field>
+      <field name="model">product.product</field>
+      <field name="inherit_id" ref="product.product_variant_easy_edit_view"/>
+      <field name="arch" type="xml">
+        <xpath expr="//group[@name='weight']" position="inside">
+          <group name="dimensions" string="Dimensions" colspan="2">
+            <field name="dimensional_uom_id"/>
+            <field name="length" attrs="{'readonly':[('type','=','service')]}"/>
+            <field name="height" attrs="{'readonly':[('type','=','service')]}"/>
+            <field name="width" attrs="{'readonly':[('type','=','service')]}"/>
+          </group>
+        </xpath>
+      </field>
+    </record>
+
+    <record model="ir.ui.view" id="product_template_only_form_view">
+      <field name="name">product_template_form_view</field>
+      <field name="model">product.template</field>
+      <field name="inherit_id" ref="product.product_template_form_view"/>
+      <field name="arch" type="xml">
+        <xpath expr="//page[@name='general_information']" position="inside">
```

Comment:
I think it makes more sense to place this group after the Logistic group.

![image](https://user-images.githubusercontent.com/1914185/61368993-67656a00-a88f-11e9-9931-ff637ca8c4c6.png)

In that case the fields would be already under the Inventory page that's hidden if the product type is service, so you can even remove the 

`attrs="{'readonly':[('type','=','service')]`